### ISSUE TITLE
fix(html-spec): allow download attribute on area element

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3587]
## Describe the bug

The `invalid-attr` rule reports the `download` attribute as disallowed on `<area>` elements. However, the HTML Living Standard allows `download` on `<area>` when the `href` attribute is present.

> https://html.spec.whatwg.org/multipage/image-maps.html#the-area-element

## Code Example

```html
<map name="m">
  <area href="/file.pdf" download alt="Download">
</map>
```

## The current behavior

```
The "download" attribute is disallowed
```

## The expected behavior

No violations reported when `href` is present.

## Impact

Causes 2 false positives in the nu-html-checker compatibility benchmark.

## Affected package

`@markuplint/html-spec` — area element spec data

**Comment by YusukeHirao** [comment]
Closing as wontfix.

The HTML Living Standard explicitly states:

> The `target`, `download`, `ping`, `rel`, and `referrerpolicy` attributes **must be omitted** if the `href` attribute is not present.
> — https://html.spec.whatwg.org/multipage/image-maps.html#the-area-element

markuplint's current behavior (reporting `download` as disallowed on `<area>` without `href`) is correct per spec. The nu-validator test file (`area/download-isvalid.html`) tests `<area download>` without `href`, which is technically a spec violation that nu-validator chooses not to enforce.
